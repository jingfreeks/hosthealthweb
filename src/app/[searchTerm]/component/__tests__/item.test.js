import { render } from "@testing-library/react";
import ItemSearch from "../item";

describe("Item Component", () => {
  it("renders Item Component to match snapshot", async () => {
    const { container } = render(
      await ItemSearch({
        result: {
          pageid: 24884283,
          ns: 0,
          title: "TeST Gliders",
          index: 3,
          thumbnail: {
            source:
              "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/104_b_aircraft_0.jpg/50px-104_b_aircraft_0.jpg",
            width: 50,
            height: 38,
          },
          pageimage: "104_b_aircraft_0.jpg",
          extract:
            "TeST Gliders was a manufacturer of ultralight sailplanes and motorgliders based in Brno, Czech Republic...",
        },
      })
    );
    expect(container).toMatchSnapshot();
  });
});
