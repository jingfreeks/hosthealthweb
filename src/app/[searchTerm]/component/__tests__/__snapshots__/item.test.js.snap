// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component renders Item Component to match snapshot 1`] = `
<div>
  <article
    class="m-3 max-w-lg"
  >
    <div
      class="flex flex-row gap-4"
    >
      <div
        class="flex flex-col justify-center"
      >
        <img
          alt="TeST Gliders"
          height="38"
          loading="lazy"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d7/104_b_aircraft_0.jpg/50px-104_b_aircraft_0.jpg"
          width="50"
        />
      </div>
      <div
        class="flex flex-col justify-center"
      >
        <h2>
          <a
            class="text-xl font-bold underline"
            href="https://en.wikipedia.org/?curid=24884283"
            target="_blank"
          >
            TeST Gliders
          </a>
        </h2>
        <p>
          TeST Gliders was a manufacturer of ultralight sailplanes and motorgliders based in Brno, Czech Republic...
        </p>
      </div>
    </div>
  </article>
</div>
`;
